llvm/lib/Target/RISCV/RISCVCodeGenPrepare.cpp

```diff
@@ -137,7 +137,8 @@ bool RISCVCodeGenPrepare::visitIntrinsicInst(IntrinsicInst &I) {
   if (expandVPStrideLoad(I))
     return true;
 
-  if (I.getIntrinsicID() != Intrinsic::vector_reduce_fadd)
+  if (I.getIntrinsicID() != Intrinsic::vector_reduce_fadd &&
+      !isa<VPReductionIntrinsic>(&I))
```

Comment:
Should this be limited to just fadd reductions?

```suggestion
      !I.getIntrinsicID() != Intrinsic::vp_reduce_fadd)
```

Reply:
Oh sorry i didn't see Craig's earlier comments, that makes sense. Can you edit the doc comment for this method to mention the VP case?